dbcon/joblist/tuple-bps.cpp

```diff
@@ -156,6 +153,109 @@ struct TupleBPSAggregators
+    }
+
+    if (fe2)
+    {
+        local_fe2Output = fe2Output;
+        local_fe2Output.initRow(&local_fe2OutRow);
+        local_fe2Data.reinit(fe2Output);
+        local_fe2Output.setData(&local_fe2Data);
+        // local_fe2OutRow = fe2OutRow;
+        local_fe2 = *fe2;
+    }
+
+    if (doJoin)
+    {
+        joinerOutput.resize(smallSideCount);
+        smallSideRows.reset(new Row[smallSideCount]);
+        smallNulls.reset(new Row[smallSideCount]);
+        smallMappings.resize(smallSideCount);
+        fergMappings.resize(smallSideCount + 1);
+        smallNullMemory.reset(new shared_array<uint8_t>[smallSideCount]);
+        local_primRG.initRow(&largeSideRow);
+        local_outputRG.initRow(&joinedBaseRow, true);
+        joinedBaseRowData.reset(new uint8_t[joinedBaseRow.getSize()]);
+        joinedBaseRow.setData(joinedBaseRowData.get());
+        joinedBaseRow.initToNull();
+        largeMapping = makeMapping(local_primRG, local_outputRG);
+
+        bool hasJoinFE = false;
+
+        for (int i = 0; i < smallSideCount; i++)
+        {
+            joinerMatchesRGs[i].initRow(&(smallSideRows[i]));
+            smallMappings[i] = makeMapping(joinerMatchesRGs[i], local_outputRG);
+
+            if (tjoiners[i]->hasFEFilter())
+            {
+                fergMappings[i] = makeMapping(joinerMatchesRGs[i], joinFERG);
+                hasJoinFE = true;
+            }
+        }
+
+        if (hasJoinFE)
+        {
+            joinFERG.initRow(&joinFERow, true);
+            joinFERowData.reset(new uint8_t[joinFERow.getSize()]);
+            memset(joinFERowData.get(), 0, joinFERow.getSize());
+            joinFERow.setData(joinFERowData.get());
+            fergMappings[smallSideCount] = makeMapping(local_primRG, joinFERG);
+        }
+
+        for (int i = 0; i < smallSideCount; i++)
+        {
+            joinerMatchesRGs[i].initRow(&(smallNulls[i]), true);
+            smallNullMemory[i].reset(new uint8_t[smallNulls[i].getSize()]);
+            smallNulls[i].setData(smallNullMemory[i].get());
+            smallNulls[i].initToNull();
+        }
+
+        local_primRG.initRow(&largeNull, true);
+        largeNullMemory.reset(new uint8_t[largeNull.getSize()]);
+        largeNull.setData(largeNullMemory.get());
+        largeNull.initToNull();
+    }
+}
+
+struct ByteStreamProcessor
+{
+    ByteStreamProcessor(TupleBPS* tbps, vector<boost::shared_ptr<messageqcpp::ByteStream>>& bsv,
+                        uint32_t begin, uint32_t end, vector<_CPInfo>& cpv, RowGroupDL* dlp,
+                        uint32_t threadID)
+        : tbps(tbps), bsv(bsv), begin(begin), end(end), cpv(cpv), dlp(dlp), threadID(threadID)
+    {
+    }
+
+    TupleBPS* tbps;
+    vector<boost::shared_ptr<messageqcpp::ByteStream>>& bsv;
+    uint32_t begin;
+    uint32_t end;
+    vector<_CPInfo>& cpv;
+    RowGroupDL* dlp;
+    uint32_t threadID;
+
+    void operator()()
+    {
+        utils::setThreadName("ByteStreamProcessor");
```

Comment:
Could you change the label so that it contains TBPS, e.g. TbpsBsProcessor, so that we know where BS processing takes place.